service/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImpl.java

```diff
@@ -25,129 +21,115 @@
 
     private final GiftCertificateDao giftCertificateDao;
     private final TagDao tagDao;
-    private final GiftCertificateTagDao giftCertificateTagDao;
-    private final GiftCertificateValidator giftCertificateValidator;
-    private final GiftCertificateDtoMapper giftCertificateDtoWrapper;
-    private final String TIME_ZONE = "GMT+3";
+    private final GiftCertificateDtoMapper giftCertificateDtoMapper;
 
 
     @Autowired
     public GiftCertificateServiceImpl(GiftCertificateDao giftCertificateDao, TagDao tagDao,
-                                      GiftCertificateTagDao giftCertificateTagDao,
-                                      GiftCertificateValidator giftCertificateValidator,
                                       GiftCertificateDtoMapper giftCertificateDtoMapper) {
         this.giftCertificateDao = giftCertificateDao;
         this.tagDao = tagDao;
-        this.giftCertificateTagDao = giftCertificateTagDao;
-        this.giftCertificateValidator = giftCertificateValidator;
-        this.giftCertificateDtoWrapper = giftCertificateDtoMapper;
+        this.giftCertificateDtoMapper = giftCertificateDtoMapper;
     }
 
 
     @Override
-    public List<GiftCertificateDto> getGiftCertificates(String tagName, String giftCertificateName, String description,
-                                                        String sortByName, String sortByDate) {
-        List<GiftCertificate> sortedGiftCertificates = giftCertificateDao.getGiftCertificates(tagName,
-                giftCertificateName, description, sortByName, sortByDate);
-
-        for (GiftCertificate giftCertificate : sortedGiftCertificates){
-            Set <Tag> tags = getTagsByGiftCertificateId(giftCertificate.getCertificateId());
-            giftCertificate.setTags(tags);
-        }
+    public List<GiftCertificateDto> findGiftCertificates(Set <String> tagsName, String giftCertificateName, String description,
+                                                        String sortByName, String sortByDate, int currentPage) {
+        Map <String,String> mapWithParameters = new HashMap<>();
+
+        mapWithParameters.put("giftCertificateName", giftCertificateName);
+        mapWithParameters.put("description", description);
+        mapWithParameters.put("sortByName", sortByName);
+        mapWithParameters.put("sortByDate", sortByDate);
+
```

Comment:
Isn't it easier to build the map in the controller where you actually get parameters and than pass the map? It will reduce method parameteres to one instead of six

Reply:
build map in the controller  and pass the map to service layer